emma-language/src/test/scala/eu/stratosphere/emma/compiler/BaseCompilerSpec.scala

```diff
@@ -43,4 +70,39 @@ trait BaseCompilerSpec extends FreeSpec with Matchers with PropertyChecks with T
     println(s"$name time: ${(System.nanoTime - s) / 1e6}ms".trim)
     ret
   }
+
+  private def wrapInClass(tree: u.Tree): u.Tree = {
+    import universe._
+    val params = api.Tree.closure(tree).map{ sym =>
+      val name = sym.name
+      val tpt = sym.typeSignature
+      q"val $name: $tpt"
+    }
+    q"""
+        class ${api.TypeName(UUID.randomUUID().toString)} {
+          def run(..$params) = {
+            $tree
+          }
+        }
+      """
+  }
 }
+
+object BaseCompilerSpec {
```

Comment:
You can use `scala-arm` (which is already in the main `pom.xml` to simplify resource handling.

1. Add the following block to `emma-language` after `<-- Scalactic -->` (in a separate `[POM]` commit).

```
        <!-- Auto-Resource Management -->
        <dependency>
            <groupId>com.jsuereth</groupId>
            <artifactId>scala-arm_${scala.tools.version}</artifactId>
            <version>${scala-arm.version}</version>
        </dependency>
```

2. Rewrite the `compileSpecPipelines` definition as follows.

```scala
object BaseCompilerSpec {

  import resource._

  val configLocation = "/test_config.properties"

  /**
   * Whether spec pipelines should try a toolbox compilation at the end, to potentially catch more bugs.
   * (Set by the IT maven profile; makes specs run about 4 times slower.)
   */
  lazy val compileSpecPipelines2 = props
    .getProperty("compile-spec-pipelines")
    .toBoolean

  lazy val props = {
    val p = new Properties()
    for {
      r <- managed(classOf[BaseCompilerSpec].getResourceAsStream(configLocation))
    } p.load(r)
    p
  }
}
```